activerecord/lib/active_record/railtie.rb

```diff
@@ -230,16 +229,6 @@ class Railtie < Rails::Railtie # :nodoc:
       end
     end
 
-    SQLITE3_PRODUCTION_WARN = "You are running SQLite in production, this is generally not recommended."\
-      " You can disable this warning by setting \"config.active_record.sqlite3_production_warning=false\"."
-    initializer "active_record.sqlite3_production_warning" do
-      if config.active_record.sqlite3_production_warning && Rails.env.production?
-        ActiveSupport.on_load(:active_record_sqlite3adapter) do
-          Rails.logger.warn(SQLITE3_PRODUCTION_WARN)
-        end
-      end
```

Comment:
Should we replace this with a deprecation warning if someone sets the value to `false`? Otherwise I worry apps will end up with a configuration line that does nothing left in their applications

Reply:
Good point, let's do it. 
I'll also work on the [docs addition](https://github.com/rails/rails/pull/49633#issuecomment-1773034807)